apps/store/src/pages/widget/[flow]/[shopSessionId]/[priceIntentId]/calculate-price.tsx

```
@@ -0,0 +1,91 @@
+  if (!isRoutingLocale(context.locale)) throw new Error(`Invalid locale: ${context.locale}`)
+
+  const serviceParams = { req: context.req, res: context.res }
+  const apolloClient = await initializeApolloServerSide(serviceParams)
+  const shopSessionService = setupShopSessionServiceServerSide({ apolloClient })
+
+  const [translations, priceIntent] = await Promise.all([
+    serverSideTranslations(context.locale),
+    fetchWidgetPriceIntent(apolloClient, context.params.priceIntentId),
+    shopSessionService.fetchById(context.params.shopSessionId),
+  ])
+
+  if (!priceIntent) return { notFound: true }
+
+  if (!isWidgetProductName(priceIntent.product.name)) {
+    console.warn(`Invalid product name: ${priceIntent.product.name}`)
+    return { notFound: true }
+  }
+
+  const priceTemplate = getPriceTemplate(priceIntent.product.name)
+
+  return addApolloState(apolloClient, {
+    props: { priceIntent, priceTemplate, ...translations, ...context.params },
+  })
+}
+
+const Page = (props: Props) => {
+  const shopSessionResult = useShopSessionQuery({
+    variables: { shopSessionId: props.shopSessionId },
+  })
+  const shopSession = shopSessionResult.data?.shopSession
+
+  const priceIntentResult = useWidgetPriceIntentQuery({
+    variables: { priceIntentId: props.priceIntentId },
+  })
+  const priceIntent = priceIntentResult.data?.priceIntent
+
+  // TODO: Handle loading state
+  if (!shopSession || !priceIntent) return null
+
+  return <CalculatePricePage {...props} priceIntent={priceIntent} />
+}
+
+export default Page
+
+const fetchWidgetPriceIntent = async (
+  apolloClient: ApolloClient<unknown>,
+  priceIntentId: string,
+) => {
+  try {
+    const { data } = await apolloClient.query<WidgetPriceIntentQuery>({
+      query: WidgetPriceIntentDocument,
+      variables: {
+        priceIntentId,
+      },
+    })
+
+    return data.priceIntent
+  } catch (error) {
+    console.error(`Failed to fetch price intent: ${priceIntentId}`, error)
+
+    return null
```

Do we need to handle this here? Why not just let it throw an error and catch in getServerSideProps?